rules/aip0192/closed_backticks.go

```diff
@@ -0,0 +1,183 @@
+			for _, line := range strings.Split(comment, "\n") {
+				// Pad the line with whitespace, so it's easier to lint
+				// backticks at the start and end of the line.
+				line = " " + line + " "
+
+				// Find all the backticks in the line and compute whether each
+				// one can be used as an opening and/or closing backtick.
+				backticks := []backtick{}
+				for _, loc := range reBacktick.FindAllStringIndex(line, -1) {
+					backticks = append(backticks, backtickAtIndex(line, loc[0]))
+				}
+
+				// Compute whether the backticks are paired.
+				backticks = computeBacktickPairs(filterUnusableBackticks(backticks))
+
+				// Add a problem for each backtick that is missing a pair.
+				for _, backtick := range backticks {
+					if backtick.pair != paired {
+						problems = append(problems, lint.Problem{
+							Message:    "Inline code should be surrounded by backticks.",
+							Suggestion: "`" + suggestionWord(line, backtick) + "`",
+							Descriptor: d,
+						})
+					}
+				}
+			}
+		}
+		return problems
+	},
+}
+
+func backtickAtIndex(line string, index int) backtick {
+	return backtick{
+		index:   index,
+		opening: separators.Contains(string(line[index-1])),
+		closing: separators.Contains(string(line[index+1])),
+	}
+}
+
+// Filter out backticks that cannot be used as opening or closing backticks.
+func filterUnusableBackticks(backticks []backtick) []backtick {
+	list := []backtick{}
+	for _, backtick := range backticks {
+		if backtick.opening || backtick.closing {
+			list = append(list, backtick)
+		}
+	}
+	return list
+}
+
+// Compute whether each backtick is paired with an adjacent backtick.
+//
+// A backtick pair consists of a set of adjacent backticks, where the first is
+// opening and the second is closing. Each backtick can be in at most one pair.
+//
+// This fills in the `pair` field for each backtick. It assumes every backtick
+// is either opening, closing, or both.
+func computeBacktickPairs(backticks []backtick) []backtick {
+	computed := []backtick{}
+
+	prevBacktickOpen := false
+	for i, backtick := range backticks {
+		backtick.pair = computePairState(backtick, prevBacktickOpen)
+
+		// If this backtick got paired, mark the previous one as paired as well.
+		if backtick.pair == paired {
+			computed[i-1].pair = paired
+		}
+
+		// If paired, this backtick cannot be used as an `opening` for another
+		// backtick.
+		prevBacktickOpen = backtick.opening && backtick.pair != paired
+
+		computed = append(computed, backtick)
+	}
+	return computed
+}
+
+func computePairState(backtick backtick, prevBacktickOpen bool) backtickPair {
+	// If the backtick is only opening, it needs a closing backtick.
+	if backtick.opening && !backtick.closing {
+		return missingClosing
+	}
+	// Otherwise, the backtick is closing. It is paired if the last backtick was
+	// opening.
+	if prevBacktickOpen {
+		return paired
+	}
+	return missingOpening
```

Comment:
One thing I realize now is that this implementation marks backticks that are both `opening` and `closing` as `missingOpening`. Perhaps a smarter way to handle this may be to assign a priority ranking among the `separators` and use that to decide whether it's `missingOpening` or `missingClosing`.

So for example, if ` ` is a stronger separator than `:`, you could have:
```
" `:thing" --> missingClosing
"thing:` " --> missingOpening
```